From: Yaroslav Halchenko <debian@onerussian.com>
Subject: BF: Initialize firstbg to the numels so if no bg value found, second loop is not executed

--- a/skimage/measure/_ccomp.pyx
+++ b/skimage/measure/_ccomp.pyx
@@ -541,7 +541,7 @@ cdef void scanBG(DTYPE_t *data_p, DTYPE_
 
     The result of this function is update of forest_p and bg parameter.
     """
-    cdef DTYPE_t i, bgval = bg.background_val, firstbg
+    cdef DTYPE_t i, bgval = bg.background_val, firstbg = shapeinfo.numels
     # We find the provisional label of the background, which is the index of
     # the first background pixel
     for i in range(shapeinfo.numels):
